src/platform/K32W/Logging.cpp

```diff
@@ -21,60 +24,56 @@
 static bool isLogInitialized;
 extern uint8_t gOtLogUartInstance;
 extern "C" void K32WWriteBlocking(const uint8_t * aBuf, uint32_t len);
+extern "C" uint32_t otPlatAlarmMilliGetNow(void);
 
 namespace chip {
 namespace Logging {
 namespace Platform {
 
-void GetMessageString(char * buf, uint8_t chipCategory, uint8_t otLevelLog)
+/* bufLen must have at least 4 (miliseconds timestamp) + 1 (category) +  */
+void GetMessageString(char * buf, uint8_t bufLen, const char * module, uint8_t category)
 {
-    if (chipCategory != kLogCategory_None)
+    int writtenLen = 0;
+    const char * categoryString;
+
+    assert(bufLen >= (TIMESTAMP_MAX_LEN_BYTES + CATEGORY_MAX_LEN_BYTES + strlen(module) + 1));
+
+    writtenLen = snprintf(buf, bufLen, "[%lu]", otPlatAlarmMilliGetNow());
+    bufLen -= writtenLen;
+    buf += writtenLen;
+
+    if (category != kLogCategory_None)
     {
-        switch (chipCategory)
+        switch (category)
         {
         case kLogCategory_Error:
-            memcpy(buf, "[Error]", 7);
+            categoryString = "E";
             break;
         case kLogCategory_Progress:
-        default:
-            memcpy(buf, "[Progress]", 10);
+            categoryString = "P";
             break;
         case kLogCategory_Detail:
-            memcpy(buf, "[Debug]", 7);
-            break;
-        }
-    }
-
-    if (otLevelLog != OT_LOG_LEVEL_NONE)
-    {
-        switch (otLevelLog)
-        {
-        case OT_LOG_LEVEL_CRIT:
-            memcpy(buf, "[Error]", 7);
-            break;
-        case OT_LOG_LEVEL_WARN:
-            memcpy(buf, "[Warn]", 6);
+            categoryString = "D";
             break;
-        case OT_LOG_LEVEL_NOTE:
-        case OT_LOG_LEVEL_INFO:
         default:
-            memcpy(buf, "[Info]", 6);
-            break;
-        case OT_LOG_LEVEL_DEBG:
-            memcpy(buf, "[Debug]", 7);
-            break;
+            categoryString = "U";
         }
+
+        writtenLen = snprintf(buf, bufLen, "[%s]", categoryString);
```

Comment:
And this would use 3 chars, not 1?

Reply:
Right, fixed this in https://github.com/project-chip/connectedhomeip/pull/8451